### ISSUE TITLE
[Feature] Support for observables in certain models

### ISSUE THREAD

**Issue by akomm** [udos86/ng2-dynamic-forms#165]
Example:
```ts
// @ng2-dynamic-forms/core/src/model/dynamic-option-control.model.d.ts
export interface DynamicOptionControlModelConfig extends DynamicFormValueControlModelConfig {
    options?: Array<DynamicFormOptionConfig>;
}
```
Could be:
```ts
export interface DynamicOptionControlModelConfig extends DynamicFormValueControlModelConfig {
    options?: Observable<Array<DynamicFormOptionConfig>>|Array<DynamicFormOptionConfig>;
}
```

This way you could attach options to a state observable e. G. using @ngrx/store which provides available options. In fact the implementation could be unified, so that when you pass a non-observable value an Observable.of(...) it used. So you do not have to handle two different cases.

This could be useful for almost everything? Things like relation which you have to impl. explicitly could be solved via obervables too, since they can project options and instantly react to changes. What requires is the binding from model to the control and the templates must bind to those observables. Currently the models are rather factory configs, less "models".

Assume you have an DynamiyFormArrayModel containing groups including `createGroup` impl. which have Select-Models - currently it gets a bit tricky to impl. this, when your options are dynamic too inside the FormArray group controls.

**Comment by udos86** [comment]
@akomm Thank you for your contribution! That's an interesting suggestion! Please give a bit of time to  evaluate this. I will let you know.

**Comment by anteburazer** [comment]
Hi everyone

I forked and modified the repository for my local project because I have customized forms and needed some extra features from this library.
Here's how I added observables for "select" control so if someone is interested: https://github.com/anteburazer/ng2-dynamic-forms/commit/b944a6700f9bc49e910fa89024be1df3f3b3f816

There's only for basic and kendo controls, but I think it's pretty similar for other libraries.
Disadvantage of this approach is that I have no solution for json import, since observable is a value and I don't know how to deserialize it.
Also I didn't implement any test for it.

This is just an idea which works for me for now.
